docs/guides/server/hostname2.adoc

```diff
@@ -0,0 +1,161 @@
+<#import "/templates/guide.adoc" as tmpl>
+<#import "/templates/kc.adoc" as kc>
+<#import "/templates/links.adoc" as links>
+
+<@tmpl.guide
+title="hostname:v2 - configuring hostname the new way"
+summary="Learn how to configure the frontend and backchannel endpoints exposed by {project_name}."
+includedOptions="hostname hostname-* proxy">
+
+== The importance of setting the hostname option
+
+By default, {project_name} mandates the configuration of the `hostname` option and does not dynamically resolve URLs. This is a security measure.
+
+{project_name} freely discloses its own URLs, for instance through the OIDC Discovery endpoint, or as part of the password reset link in an email. If the hostname was dynamically interpreted from a hostname header, it could provide a potential attacker with an opportunity to manipulate a URL in the email, redirect a user to the attacker's fake domain, and steal sensitive data such as action tokens, passwords, etc.
+
+By explicitly setting the `hostname` option, we avoid a situation where tokens could be issued by a fraudulent issuer. The server can be started with an explicit hostname using the following command:
+
+[source,bash]
+----
+kc.sh start --hostname my.keycloak.org
```

Comment:
We should be using `<@kc.start ...` instead of hardcoded `kc.sh`.

Reply:
Addressed.